src/ConnectedMode/IssueMatcher.cs

```diff
@@ -0,0 +1,94 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using SonarLint.VisualStudio.Core.Helpers;
+using SonarLint.VisualStudio.Core.Suppressions;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.ConnectedMode
+{
+    internal interface IIssueMatcher
+    {
+        bool IsGoodMatch(IFilterableIssue issue, SonarQubeIssue serverIssue);
```

Comment:
A name which describes better should be used IMO. When I read the name of the method I do not understand what is a "good match" I need to look into the method to understand it. Something like "IsMatchWithFilePath" would be better.

Reply:
My idea was to indicate that it's not a 100% certain match (ie it can have both FPs and FNs). Maybe you have other suggestions considering my initial intention?

Reply:
then IsProbableMatch maybe? Good is too vague for me.

Reply:
Or at least add some documentation comments